lib/phoenix_html/link.ex

```diff
@@ -165,4 +175,32 @@ defmodule Phoenix.HTML.Link do
 
     {value, opts}
   end
+
+  defp valid_destination!(to, calling_func) do
+    if invalid_destination?(to) do
+      raise ArgumentError, """
+      unsupported scheme given to #{calling_func}. In case you want to link to an
+      unknown or unsafe scheme, such as javascript, use a tuple: {:javascript, rest}
+      """
+    end
+
+    case to do
+      {_, dest} -> dest
+      dest -> dest
+    end
+  end
+
+  for scheme <- @valid_uri_schemes do
+    defp invalid_destination?(unquote(scheme) <> _), do: false
+  end
+  defp invalid_destination?({:safe, to}) do
+    invalid_destination?(to)
+  end
+  defp invalid_destination?({scheme, to}) when is_atom(scheme) do
+    !String.starts_with?(to, "#{scheme}:")
```

Comment:
Tiniest nitpick ever: we prefer to use `not` instead of `!` when the result is guaranteed to be a boolean. :)

Reply:
Question: for an unsupported prefix, what if we automatically add the prefix instead of checking if it is the prefixed being used? So I would pass: `{:javascript, "alert(0)"}`. Wouldn't that be cleaner?

Reply:
Sure! Auto-prefix makes sense. My thought was that this would be an opportunity to validate that the URI conforms, but prefixing and ensuring validity of "to" should accomplish the same thing. And, you're right, it does feel cleaner.